runtime/interpreter/primitivestatictype.go

```diff
@@ -186,6 +186,82 @@ const (
 
 func (PrimitiveStaticType) isStaticType() {}
 
+func (t PrimitiveStaticType) elementSize() uint {
+	switch t {
+	case
+		PrimitiveStaticTypeAnyStruct,
+		PrimitiveStaticTypeAnyResource,
+		PrimitiveStaticTypeAny:
+		return UnknownElementSize
+	case PrimitiveStaticTypeVoid:
+		return 3 // see VoidValue ByteLength
+	case PrimitiveStaticTypeNever:
+		return 1
+	case PrimitiveStaticTypeBool:
+		return 1
+	case PrimitiveStaticTypeAddress:
+		return 8 // address length is 8 bytes
+	case PrimitiveStaticTypeString,
+		PrimitiveStaticTypeCharacter,
+		PrimitiveStaticTypeMetaType,
+		PrimitiveStaticTypeBlock:
+		return UnknownElementSize
+
+	case PrimitiveStaticTypeFixedPoint,
+		PrimitiveStaticTypeSignedFixedPoint:
+		return 8
+
+	// values of these types may wrap big.Int
+	case PrimitiveStaticTypeInt,
+		PrimitiveStaticTypeUInt,
+		PrimitiveStaticTypeUInt128,
+		PrimitiveStaticTypeUInt256,
+		PrimitiveStaticTypeInt128,
+		PrimitiveStaticTypeInt256,
+		PrimitiveStaticTypeInteger,
+		PrimitiveStaticTypeSignedInteger,
+		PrimitiveStaticTypeNumber,
+		PrimitiveStaticTypeSignedNumber:
+		return UnknownElementSize
+
+	case PrimitiveStaticTypeInt8,
+		PrimitiveStaticTypeUInt8,
+		PrimitiveStaticTypeWord8:
+		return 1
+	case PrimitiveStaticTypeInt16,
+		PrimitiveStaticTypeUInt16,
+		PrimitiveStaticTypeWord16:
+		return 2
+	case PrimitiveStaticTypeInt32,
+		PrimitiveStaticTypeUInt32,
+		PrimitiveStaticTypeWord32:
+		return 4
+	case PrimitiveStaticTypeInt64,
+		PrimitiveStaticTypeUInt64,
+		PrimitiveStaticTypeWord64,
+		PrimitiveStaticTypeFix64,
+		PrimitiveStaticTypeUFix64:
+		return 8
```

Comment:
This is an estimation of the number of values of each type that can be packed into one (1024 byte) atree data slab, the idea being that we can use this information and the atree's `Count` in order to get a more accurate estimation of the number of data slabs in an atree. cc @fxamacker are the values stored in an atree `cbor` encoded; do we need to add the tag size here?

Reply:
Thank you for the clarification, adding a comment, and adjusting the values.

Last concern I have is that the `StaticType.elementSize` and `Value.ByteLength` functions become "out-of-sync". They return different values (the first returns the max, the second is based on the concrete value), but changing the latter could still cause the former to become outdated / under-estimating. 

It is hard to prevent this, but maybe a warning/remding comment in the `Value.ByteLength` implementations could help, e.g. `// NOTE: ensure to keep StaticType.elementSize in sync!`. Please feel free to add this in a follow-up PR, it's not blocking this one